cmd/juju/cloud/add.go

```diff
@@ -109,6 +110,9 @@ func (c *AddCloudCommand) Init(args []string) (err error) {
 		c.Cloud = args[0]
 	}
 	if len(args) > 1 {
+		if c.CloudFile != args[1] && c.CloudFile != "" {
```

Comment:
The intent is exactly what you describe: to error if `CloudFile` has been provided twice, but isn't the same. For justification, please see my earlier comment.